### PR TITLE
Add PyTorch Profiler

### PR DESCRIPTION
**Pull request by stu1130**
## Channel for questions ##
Before or while filing an issue please feel free to join our [Slack channel](https://join.slack.com/t/deepjavalibrary/shared_invite/zt-ar91gjkz-qbXhr1l~LFGEIEeGBibT7w) to get in touch with development team, ask questions, find out what's cooking and more!

## Description ##
(Brief description of what this PR is about)

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- [ ] Code is well-documented: 
    - For user-facing API changes, Java doc has been updated. 
    - For new examples, README.md is added to explain the what the example does.
- [ ] To the my best knowledge, [examples](https://github.com/awslabs/djl/tree/master/examples) and [jupyter notebooks](https://github.com/awslabs/djl/tree/master/jupyter) are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, Java doc)
- [ ] Feature2, tests, (and when applicable, Java doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made?
- Interesting edge cases to note here

### CODE REVIEW

File: pytorch/pytorch-engine/src/test/java/integration/MkldnnTest.java [hunk 1]
```diff
@@ -46,5 +46,6 @@ public void testMKLDNNForPyTorch() {
             Arrays.stream(arrays).reduce(NDArray::add);
             Arrays.stream(arrays).forEach(NDArray::toString);
         }
+        System.setProperty("ai.djl.pytorch.use_mkldnn", "false");
```
Comment (frankfliu) [inline pytorch/pytorch-engine/src/test/java/integration/MkldnnTest.java:49]: Add to finally section

File: pytorch/pytorch-engine/src/main/java/ai/djl/pytorch/jni/JniUtils.java [hunk 2]
```diff
@@ -90,6 +90,14 @@ public static void setSeed(long seed) {
         PyTorchLibrary.LIB.torchManualSeed(seed);
     }
 
+    public static Pointer createRecordProfiler(String name) {
```
Comment (frankfliu) [inline pytorch/pytorch-engine/src/main/java/ai/djl/pytorch/jni/JniUtils.java:93]: I think it's better make it simple:

```
public static synchronized void startProfile(boolean profileMemory, boolean recordShape, boolean useCuda) {
}
public static synchronized void stopProfile() {
}
```

File: api/src/main/java/ai/djl/modality/cv/BufferedImageFactory.java [hunk 3]
```diff
@@ -95,8 +95,8 @@ public Image fromNDArray(NDArray array) {
         if (shape.dimension() != 3) {
             throw new IllegalArgumentException("Shape should only have three dimension follow CHW");
         }
-        if (array.getDataType() != DataType.UINT8 && array.getDataType() != DataType.INT8) {
-            throw new IllegalArgumentException("Datatype should be INT8");
+        if (array.getDataType() != DataType.UINT8) {
```
Comment (frankfliu) [inline api/src/main/java/ai/djl/modality/cv/BufferedImageFactory.java:98]: INT8 should work, why we block it?

File: pytorch/pytorch-engine/src/main/java/ai/djl/pytorch/jni/JniUtils.java [hunk 4]
```diff
@@ -90,6 +92,22 @@ public static void setSeed(long seed) {
         PyTorchLibrary.LIB.torchManualSeed(seed);
     }
 
+    public static synchronized void startProfile(String name) {
```
Comment (frankfliu) [inline pytorch/pytorch-engine/src/main/java/ai/djl/pytorch/jni/JniUtils.java:95]: Can you add a JavaDoc here?

And change the name to String outputFile

Comment (frankfliu) [reply pytorch/pytorch-engine/src/main/java/ai/djl/pytorch/jni/JniUtils.java:95]: Why python has 3 options (shape, memory, cuda) for profile?

**Review by frankfliu — APPROVED** [review]

**Comment by stu1130** [comment]
It seems the output shape and memory usage info are still not showing up. investigating

**Comment by stu1130** [comment]
figured out why, will raise another PR to add support for it
